app/models/payment.rb

```diff
@@ -32,34 +32,34 @@ class Payment < ApplicationRecord
   belongs_to :user
 
   scope :current_camp_payments, -> { where('camp_year = ? ', CampConfiguration.active_camp_year) }
-  scope :status1_current_camp_payments, -> { current_camp_payments.where('transaction_status = ?', "1") }
+  scope :status1_current_camp_payments, -> { current_camp_payments.where('transaction_status = ?', '1') }
 
   private
 
   def set_status
-    return unless self.transaction_status == '1'
+    return unless transaction_status == '1'
 
-    @current_enrollment = self.user.enrollments.current_camp_year_applications.last
-    if self.user.payments.current_camp_payments.where(transaction_status: 1).count == 1
-      RegistrationMailer.app_complete_email(self.user).deliver_now
-      @current_enrollment.update!(application_status: "submitted", application_status_updated_on: Date.today)
-      if @current_enrollment.recommendation.present?
-        if @current_enrollment.recommendation.recupload.present? 
-          @current_enrollment.update!(application_status: "application complete", application_status_updated_on: Date.today)
-        end
-      end
-    else 
-      if balance_due == 0 && @current_enrollment.camp_doc_form_completed
-        @current_enrollment.update!(application_status: "enrolled", application_status_updated_on: Date.today)
+    @current_enrollment = user.enrollments.current_camp_year_applications.last
+
+    # Check if this is the first payment (when required) or if no payment was required
+    if !@current_enrollment.application_fee_required || user.payments.current_camp_payments.where(transaction_status: 1).count == 1
+      RegistrationMailer.app_complete_email(user).deliver_now
+      @current_enrollment.update!(application_status: 'submitted', application_status_updated_on: Date.today)
```

Comment:
The application status is not changing to 'submitted' if application_fee_required is false because an applicant is not paying the fee and the app is not going to the Payment model.
And it looks like the app_complete_email email is not sent to the applicant.

Reply:
The @current_enrollment.application_fee_required here is the one set in the enrollment record not the CampConfiguration record. 

The current_enrollment.application_fee_required is set to false when a new enrollment is created and the CampConfiguration is not requiring the application fee. 

Be sure you're testing an enrollment that is created properly.

Reply:
I tested properly. I set up application_fee_required in the CampConfiguration and submitted a new application.
The application status was not changed to 'submitted' and the 'Confirmation of application' email was not sent.
The above actions happen in the Payment model, and if an applicant is not paying an application fee, that would not happen until the first payment.

Reply:
Did the same test that I described in my previous comment work for you?

Reply:
I see what was wrong. The Payment being created was managing the status update and mailer sending. I twiggled this so the recommendation creation will take care of sending the RegistrationMailer if the require application fee is set for the enrollment record